src/bin/pg_autoctl/pgsql.c

```diff
@@ -1640,8 +1640,8 @@ pgsql_replication_slot_create_and_drop(PGSQL *pgsql, NodeAddressArray *nodeArray
 	if (!BuildNodesArrayValues(nodeArray, &sqlParams, values))
 	{
 		/* errors have already been logged */
-		PQfreemem(query);
-		PQfreemem(values);
+		destroyPQExpBuffer(query);
+		destroyPQExpBuffer(values);
```

Comment:
This same issue seems to be present in the function `pgsql_replication_slot_maintain`

Reply:
Fixed now. Curiously didn't appear in the valgrind report.